python_frank_energie/models.py

```
@@ -264,82 +1468,933 @@ def today(self) -> list[Price]:
+    def today_tax_markup_avg(self) -> float:
+        """ Average market price including tax and markup for today. """
+        today_market_tax_markup = [
+            hour.marketPrice_with_tax_and_markup for hour in self.today_prices]
+        return mean(today_market_tax_markup)
 
     @property
-    def today_min(self) -> Price:
+    def today_min(self) -> Optional[Price]:
         """Price with the lowest total for today."""
-        return min([hour for hour in self.today], key=lambda hour: hour.total)
+        if not self.today == []:
```

Ruff

Use `self.today != []` instead of `not self.today == []`

Replace with `!=` operator

(SIM201)